packages/builder/src/components/design/settings/controls/GridColumnConfiguration/PrimaryColumnFieldSetting.svelte

```diff
@@ -0,0 +1,101 @@
+<script>
+  import EditComponentPopover from "../EditComponentPopover.svelte"
+  import { Icon } from "@budibase/bbui"
+  import { setContext } from "svelte"
+  import { writable } from "svelte/store"
+  import { FieldTypeToComponentMap } from "../FieldConfiguration/utils"
+  import { store } from "builderStore"
+
+  export let item
+  export let anchor
+
+  let draggableStore = writable({
+    selected: null,
+    actions: {
+      select: id => {
+        draggableStore.update(state => ({
+          ...state,
+          selected: id,
+        }))
+      },
+    },
+  })
+
+  setContext("draggable", draggableStore)
+
+  const parseSettings = settings => {
+    return settings
+      .filter(setting => setting.key !== "field")
+      .map(setting => {
+        return { ...setting, nested: true }
+      })
+  }
+
+  const getIcon = () => {
+    const component = `@budibase/standard-components/${
+      FieldTypeToComponentMap[item.columnType]
+    }`
+    return store.actions.components.getDefinition(component).icon
```

Comment:
This should obviously never be null, but we should still add optional chaining to ensure we wouldn't crash the page if we couldn't find the definition.